test/cpp/util/cli_credentials.cc

```diff
@@ -49,7 +56,40 @@ std::shared_ptr<grpc::ChannelCredentials> CliCredentials::GetCredentials()
     if (FLAGS_use_auth) {
```

Comment:
Some flags combinations are still meaningless and error prone. I would have a helper function like:

```cpp
void CheckForInvalidSslFlags(const string& flag) {
  if (!FLAGS_ssl_cleint_cert_chain_file.empty() || !FLAGS_...) {
    std::cerr << "SSL specific flags cannot be used with " << flag << " flag." << std::endl;
    exit(2);
  }
}
```
and call this from:

```cpp
if (!FLAGS_enable_ssl) {
  CheckForInvalidSslFlags("--noenable_ssl");
  return grpc::InsecureCredentials();
} else {
  if (FLAGS_use_auth) {
    CheckForInvalidSslFlags("--use_auth");
    return grpc::GoogleDefaultCredentials();
  } 
...
```